openverse_catalog/dags/providers/provider_api_scripts/smithsonian.py

```diff
@@ -1,531 +1,443 @@
+        online_media = self._check_type(dnr_dict.get("online_media"), dict)
+        return self._check_type(online_media.get("media"), list)
+
+    @staticmethod
+    def _process_image_list(image_list, partial_image_data: dict) -> list:
+        images = []
+        for image_data in image_list:
+            usage = image_data.get("usage", {}).get("access")
+            if image_data.get("type") != "Images" or usage != "CC0":
+                continue
+
+            if (image_url := image_data.get("content")) is None:
+                continue
+
+            if (foreign_identifier := image_data.get("idsId")) is None:
+                continue
+
+            images.append(
+                {
+                    **partial_image_data,
+                    "image_url": image_url,
+                    "foreign_identifier": foreign_identifier,
+                }
+            )
+        return images
+
+    def _get_foreign_landing_url(self, row):
+        logger.debug("Getting foreign_landing_url from row")
+        dnr_dict = self._get_descriptive_non_repeating_dict(row)
+        foreign_landing_url = dnr_dict.get("record_link")
+        if foreign_landing_url is None:
+            foreign_landing_url = dnr_dict.get("guid")
+
+        return foreign_landing_url
+
+    def _get_freetext_dict(self, row):
+        logger.debug("Getting freetext_dict from row")
+        freetext = self._get_content_dict(row).get("freetext")
+        return self._check_type(freetext, dict)
+
+    def _extract_meta_data(self, row) -> dict:
+        freetext = self._get_freetext_dict(row)
+        descriptive_non_repeating = self._get_descriptive_non_repeating_dict(row)
+        description, label_texts = "", ""
+        notes = self._check_type(freetext.get("notes"), list)
+
+        for note in notes:
+            label = self._check_type(note.get("label", ""), str)
+            if label.lower().strip() in self.description_types:
+                description += " " + self._check_type(note.get("content", ""), str)
+            elif label.lower().strip() == "label text":
+                label_texts += " " + self._check_type(note.get("content", ""), str)
+
+        meta_data = {
+            "unit_code": descriptive_non_repeating.get("unit_code"),
+            "data_source": descriptive_non_repeating.get("data_source"),
+        }
+
+        if description:
+            meta_data.update(description=description.strip())
+        if label_texts:
+            meta_data.update(label_text=label_texts.strip())
+
+        return {k: v for (k, v) in meta_data.items() if v is not None}
+
+    def _extract_source(self, meta_data):
+        unit_code = meta_data.get("unit_code").strip()
+        source = next(
+            (s for s in self.sub_providers if unit_code in self.sub_providers[s]), None
+        )
+        if source is None:
+            raise Exception(f"An unknown unit code value {unit_code} encountered ")
+        return source
+
+    def _get_creator(self, row):
```

Comment:
This is wild 😮 I know this is all from the original code, but it would be great to comment this section if we have any more context (not necessarily now).